// fauna-schema-migrate/dist/index.js
"use strict";
// Copyright Fauna, Inc.
// SPDX-License-Identifier: MIT-0
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateRollbackQuery = exports.generateApplyQuery = exports.generateMigrationLetObject = exports.generateMigrations = exports.diffSnippets = exports.getSnippetsFromPaths = exports.getSnippetsFromCode = exports.getSnippetsFromStrings = void 0;
var from_code_1 = require("./state/from-code");
Object.defineProperty(exports, "getSnippetsFromStrings", { enumerable: true, get: function () { return from_code_1.getSnippetsFromStrings; } });
Object.defineProperty(exports, "getSnippetsFromCode", { enumerable: true, get: function () { return from_code_1.getSnippetsFromCode; } });
Object.defineProperty(exports, "getSnippetsFromPaths", { enumerable: true, get: function () { return from_code_1.getSnippetsFromPaths; } });
var diff_1 = require("./migrations/diff");
Object.defineProperty(exports, "diffSnippets", { enumerable: true, get: function () { return diff_1.diffSnippets; } });
var generate_migration_1 = require("./migrations/generate-migration");
Object.defineProperty(exports, "generateMigrations", { enumerable: true, get: function () { return generate_migration_1.generateMigrations; } });
var generate_query_1 = require("./migrations/generate-query");
Object.defineProperty(exports, "generateMigrationLetObject", { enumerable: true, get: function () { return generate_query_1.generateMigrationLetObject; } });
var advance_1 = require("./migrations/advance");
Object.defineProperty(exports, "generateApplyQuery", { enumerable: true, get: function () { return advance_1.generateApplyQuery; } });
var rollback_1 = require("./migrations/rollback");
Object.defineProperty(exports, "generateRollbackQuery", { enumerable: true, get: function () { return rollback_1.generateRollbackQuery; } });
