// @next-auth/typeorm-legacy-adapter/dist/models/user.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UserSchema = exports.User = void 0;

class User {
  constructor(name, email, image, emailVerified) {
    if (name) {
      this.name = name;
    }

    if (email) {
      this.email = email;
    }

    if (image) {
      this.image = image;
    }

    if (emailVerified) {
      const currentDate = new Date();
      this.emailVerified = currentDate;
    }
  }

}

exports.User = User;
const UserSchema = {
  name: "User",
  target: User,
  columns: {
    id: {
      primary: true,
      type: "int",
      generated: true
    },
    name: {
      type: "varchar",
      nullable: true
    },
    email: {
      type: "varchar",
      unique: true,
      nullable: true
    },
    emailVerified: {
      type: "timestamp",
      nullable: true
    },
    image: {
      type: "varchar",
      nullable: true
    },
    createdAt: {
      type: "timestamp",
      createDate: true
    },
    updatedAt: {
      type: "timestamp",
      updateDate: true
    }
  }
};
exports.UserSchema = UserSchema;
